gobblin-utility/src/main/java/org/apache/gobblin/util/jdbc/DataSourceProvider.java

```diff
@@ -81,6 +83,7 @@ public DataSourceProvider(@Named("dataSourceProperties") Properties properties)
   }
 
   public DataSourceProvider() {
+    log.warn("Creating {} without setting validation query", this.getClass().getSimpleName());
```

Comment:
another great idea.  might consider including `Thread.currentThread().getStackTrace()` to better pinpoint context.